src/stats/cdf.rs

```diff
@@ -0,0 +1,211 @@
+// Copyright 2016 Johannes Köster.
+// Licensed under the MIT license (http://opensource.org/licenses/MIT)
+// This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! Support for discrete probability distributions in terms of cumulative distribution
+//! functions (CDF).
+
+use std::f64;
+use std::iter;
+use std::slice;
+
+use num::traits::{cast, NumCast};
+use itertools::Itertools;
+
+use stats::logprobs::{self, LogProb};
+
+
+/// Implementation of a cumulative distribution function.
+#[derive(Debug, Clone)]
+pub struct CDF<T: PartialOrd> {
+    inner: Vec<(T, LogProb)>
+}
+
+
+impl<T: PartialOrd> CDF<T> {
+    /// Create CDF from given probability mass function (PMF). The PMF may contain duplicate values
+    /// the probabilities of which are summed during generation of the CDF.
+    ///
+    /// # Arguments
+    ///
+    /// * `pmf` - the PMF as a vector of value/probability pairs
+    pub fn from_pmf(mut entries: Vec<(T, LogProb)>) -> Self {
+        entries.sort_by(|&(ref a, _), &(ref b, _)| a.partial_cmp(b).unwrap());
```

Comment:
You are somehow right. I unwrap it in order to allow floats for T. In case of NaN, this would panic. One could think about returning a result instead. What's your feeling? Panic or result?